lib/spack/spack/binary_distribution.py

```diff
@@ -712,6 +722,14 @@ def get_buildfile_manifest(spec):
     root = spec.prefix
     visit_directory_tree(root, visitor)
 
+    # Collect a list of prefixes for this package and it's dependencies, Spack will
+    # look for them to decide if text file needs to be relocated or not
+    prefixes = [d.prefix for d in spec.traverse(root=False, deptype="all") if not d.external]
+    prefixes.append(spec.prefix)
```

Comment:
```suggestion
    prefixes = [d.prefix for d in spec.traverse(root=True, deptype="all") if not d.external]
    prefixes.append(spack.hooks.sbang.sbang_install_path)
```

You need to include the sbang path, since that's moved too.